arangod/Agency/Store.cpp

```diff
@@ -692,7 +692,41 @@ bool Store::applies(arangodb::velocypack::Slice const& transaction) {
     Slice value = transaction.get(key);
 
     if (value.isObject() && value.hasKey("op")) {
-      _node.hasAsWritableNode(abskeys.at(i)).first.applieOp(value);
+      if (value.get("op").isEqualString("delete") ||
+          value.get("op").isEqualString("replace") ||
+          value.get("op").isEqualString("erase")) {
+        if (!_node.has(abskeys.at(i))) {
+          continue;
+        }
+      } 
+      auto uri = abskeys.at(i);
+      if (value.get("op").isEqualString("observe")) {
+        if (value.hasKey("url") && value.get("url").isString()) {
+          auto url = value.get("url").copyString();
+          _observerTable.emplace(std::pair<std::string, std::string>(url, uri));
+          _observedTable.emplace(std::pair<std::string, std::string>(uri, url));
+        }
+      } else if (value.get("op").isEqualString("unobserve")) {
+        if (value.hasKey("url") && value.get("url").isString()) {
+          auto url = value.get("url").copyString();
+          auto ret = _observerTable.equal_range(url);
+          for (auto it = ret.first; it != ret.second; ++it) {
+            if (it->second == uri) {
+              _observerTable.erase(it);
+              break;
+            }
+          }
+          ret = _observedTable.equal_range(uri);
+          for (auto it = ret.first; it != ret.second; ++it) {
+            if (it->second == url) {
+              _observedTable.erase(it);
+              break;
+            }
+          }
+        }
```

Comment:
Can we maybe create two methods: `observe` and `unobserve` which handles this logic at a single place?

Reply:
not with this PR. let's not accidentally break things

Reply:
and the logic is somewhat different.